hack/update-codegen.sh

```diff
@@ -29,120 +29,85 @@ while getopts "v" opt; do
+    "$KF_PACKAGE/pkg/client/build" \
+    "github.com/knative/build/pkg/apis" \
+    "$BUILD_RESOURCE" \
+    --go-header-file="$HEADER_FILE" \
+    "${GENERATOR_FLAGS}"
+  [ $? -ne 0 ] && echo Error running code-generator 1>&2 && exit 1
 }
 
-for PACKAGE in ${CODEGEN_PACKAGES}; do
-  if [ ! -d $(go env GOPATH)/src/${PACKAGE} ]; then
-    install_package $PACKAGE
-  fi
-done
-
-if [ "$GOPATH/src/github.com/google/kf" != "$(git rev-parse --show-toplevel)" ]; then
-    echo "The generator scripts aren't go module compatible (yet)." 1>&2
-    exit 1
-fi
-
-# Fix issues due to using old k8s.io/client-go
-# The generator wants to use a codec that is only available in a version of
-# k8s.io/apimachinery that we can't yet use.
-os_friendly_sed () {
-  echo "Applying $1 to $2"
-  if [ ! -e "$2" ]; then
-    echo file not found: $2 2>&1
-    exit 1
-  fi
-  sed "$1" "$2" > "$2.new"
-  mv "$2.new" "$2"
+knative-injection-gen() {
+  echo running knative-injection-generator
+  commit=$(cat go.mod | grep knative.dev/pkg | tr '-' ' ' | awk '{print $NF}')
+  curl -LOJ https://raw.githubusercontent.com/knative/pkg/${commit}/hack/generate-knative.sh
+  chmod +x generate-knative.sh
+  mkdir -p vendor/knative.dev/pkg/hack
+  mv generate-knative.sh ./vendor/knative.dev/pkg/hack
+  # Do Knative injection generation
+  bash -x ${KNATIVE_CODEGEN_PKG}/hack/generate-knative.sh \
+    "injection" \
+    "github.com/google/kf/pkg/client" \
+    "github.com/google/kf/pkg/apis" \
+    "kf:v1alpha1" \
+    --go-header-file $HEADER_FILE
+
+  bash -x ${KNATIVE_CODEGEN_PKG}/hack/generate-knative.sh \
+    "injection" \
+    "github.com/google/kf/pkg/client/build" \
+    "github.com/knative/build/pkg/apis" \
+    "build:v1alpha1" \
+    --go-header-file $HEADER_FILE
 }
 
-${CODEGEN_PKG}/generate-groups.sh all \
-  "$KF_PACKAGE/pkg/client" \
-  "$KF_PACKAGE/pkg/apis" \
-  "$KF_RESOURCE" \
-  --go-header-file="$HEADER_FILE" \
-  "${GENERATOR_FLAGS}"
-[ $? -ne 0 ] && echo Error running code-generator 1>&2 && exit 1
-
-${CODEGEN_PKG}/generate-groups.sh "deepcopy,client,informer,lister" \
-  "$KF_PACKAGE/pkg/client/build" \
-  "github.com/knative/build/pkg/apis" \
-  "$BUILD_RESOURCE" \
-  --go-header-file="$HEADER_FILE" \
-  "${GENERATOR_FLAGS}"
-[ $? -ne 0 ] && echo Error running code-generator 1>&2 && exit 1
-
-codegen_sed() {
-  local CLIENT=$1
-  local TYPE=$2
-  local VERSION=$3
-  local FILE=$4
-
-  if [ "$CLIENT" = "kf" ]; then
-    CLIENT=""
-  fi
-
-  os_friendly_sed 's/scheme.Codecs.WithoutConversion()/scheme.Codecs/g' "$KF_PACKAGE_LOCATION/pkg/client/$CLIENT/clientset/versioned/typed/$TYPE/$VERSION/$FILE.go"
-  os_friendly_sed 's/pt, //g' "$KF_PACKAGE_LOCATION/pkg/client/$CLIENT/clientset/versioned/typed/$TYPE/$VERSION/fake/fake_${FILE}.go"
-}
+GO111MODULE=on
+go mod vendor
 
-TYPES="build_client build buildtemplate clusterbuildtemplate"
-for type in ${TYPES}; do
-  codegen_sed build build v1alpha1 $type
-done
+case $GENS in
```

Comment:
I think with this change, check-code-generator.sh needs to be udpated

Reply:
If nothing is supplied, all is assumed